### PR TITLE
Store screenshot test output as Cirrus artifacts; do fuzzy comparison of non-matching screenshot pixels

### PR DESCRIPTION
**Pull request by yjbanov**
- Store screenshot test output as Cirrus artifacts
- Do fuzzy comparison of non-matching screenshot pixels
- Make screenshot tests fail on Cirrus

### CODE REVIEW

File: lib/web_ui/dev/goldens.dart [hunk 1]
```diff
@@ -33,10 +42,60 @@ class ImageDiff {
...
+  static const double _kColorDistanceThreshold = 0.1;
+
   final int _colorOk = Color.fromRgb(255, 255, 255);
   final int _colorBadPixel = Color.fromRgb(255, 0, 0);
   final int _colorExpectedPixel = Color.fromRgb(0, 255, 0);
 
+  static int _reflectedPixel(Image image, int x, int y) {
+    x = x.abs();
+    if (x == image.width) {
+      x = image.width - 2;
```
Comment (mdebbar) [inline lib/web_ui/dev/goldens.dart]: Maybe a comment on what this does and why it's necessary?

Comment (yjbanov) [reply lib/web_ui/dev/goldens.dart]: Done. Added a dart doc explaining what "reflecting" means.

**Comment by dnfield** [comment]
For reasons entirely unclear, this failed on LUCI building ios_debug_sim.

File: lib/web_ui/test/golden_files/.gitignore [hunk 2]
```diff
@@ -1,3 +0,0 @@
-# Failed screenshot files
```
Comment (ditman) [inline lib/web_ui/test/golden_files/.gitignore]: I think we should ignore the test output artifacts so people don't add them by mistake to the repo. Or maybe we can have a `felt goldens:clean` command that removes all the junk that failing (or almost-failing) tests generate.

Comment (ditman) [reply lib/web_ui/test/golden_files/.gitignore]: (Other than that, LGTM!)

Comment (yjbanov) [reply lib/web_ui/test/golden_files/.gitignore]: I thought my change stopped writing to the `test/` directory, but let me double-check.

Comment (ditman) [reply lib/web_ui/test/golden_files/.gitignore]: You're right, it's writing to test_results within the dart tool. DISREGARD!

**Pull request merged**